plugins/kubernetes/app/assets/javascripts/kubernetes/controllers/kubernetes_create_release_ctrl.js

```diff
@@ -0,0 +1,206 @@
+        $scope.roles = [];
+        $scope.deploy_groups = [];
+      }
+      else {
+        // Select only the deploy groups associated with a kubernetes cluster
+        $scope.deploy_groups = _.filter($scope.environment.deploy_groups, function(deploy_group) {
+          return _.isNotUndefinedOrEmpty(deploy_group.kubernetes_cluster);
+        });
+
+        loadRoles().then(
+          function(roles) {
+            _.each($scope.deploy_groups, function(deploy_group) {
+              deploy_group.roles = _.cloneArray(roles);
+            });
+          }
+        );
+      }
+    };
+
+    $scope.toggleAll = function() {
+      if ($scope.allToggled()) {
+        _.each($scope.deploy_groups, function(deploy_group) {
+          deploy_group.selected = false;
+        });
+      }
+      else {
+        _.each($scope.deploy_groups, function(deploy_group) {
+          deploy_group.selected = true;
+        });
+      }
+    };
+
+    $scope.allToggled = function() {
+      return _.every($scope.deploy_groups, function(deploy_group) {
+        return deploy_group.selected;
+      });
+    };
+
+    $scope.toggleSelection = function(deploy_group) {
+      deploy_group.selected = !deploy_group.selected;
+    };
+
+    $scope.isSelected = function(deploy_group) {
+      return deploy_group.selected;
+    };
+
+    $scope.validate = function(step) {
+      var valid = true;
+      valid &= _.isDefined($scope.environment);
+      valid &= _.isDefined($scope.build);
+      valid &= _.isNotUndefinedOrEmpty($scope.roles);
+
+      if (step == 2) {
+        valid &= _.isNotUndefinedOrEmpty($scope.deploy_groups)
+      }
+
+      return valid;
+    };
+
+    $scope.submit = function() {
+      createRelease();
+    };
+
+    /*
+     Controller private functions.
+     */
+
+    function loadEnvironments() {
+      $scope.loadingEnvironments = true;
+      environmentsService.loadEnvironments($stateParams.project_id).then(
+        function(environments) {
+          $scope.environments = environments;
+          $scope.loadingEnvironments = false;
+        },
+        function(result) {
+          handleFailure(result);
+        }
+      );
+    }
+
+    function loadBuilds() {
+      $scope.loadingBuilds = true;
+      buildsService.loadBuilds($stateParams.project_id).then(
+        function(builds) {
+          $scope.builds = builds.map(buildFactory.build);
+          $scope.loadingBuilds = false;
+        },
+        function(result) {
+          handleFailure(result);
+        }
+      );
+    }
+
+    function loadRoles() {
+      $scope.roles = undefined;
+      $scope.loadingRoles = true;
+
+      var deferred = $q.defer();
+      kubernetesService.loadRoles($stateParams.project_id).then(
+        function(roles) {
+          $scope.loadingRoles = false;
+          $scope.roles = roles;
+
+          //todo: merge with environment defaults
+
+          deferred.resolve(roles);
+        },
+        function(result) {
```

Comment:
Isn't the error case already handled in the Service itself? Notifying the error?